### PR TITLE
Add SetSeverity to logging interface

### PR DESCRIPTION
**Pull request by pquerna**
This should allow runtime changing of the Severity levels, which is blocking upgrading the package for vulcand:

https://github.com/mailgun/vulcand/issues/211

### CODE REVIEW

**Comment by jeremyschlatter** [comment]
```
$ go test
# github.com/mailgun/log
./log_test.go:25: cannot use newTestLogger("log1") (type *testLogger) as type Logger in argument to Init:
	*testLogger does not implement Logger (missing SetSeverity method)
```

File: console.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ func (l *writerLogger) Writer(sev Severity) io.Writer {
 	return nil
 }
 
+func (l *writerLogger) SetSeverity(sev Severity) {
+	l.sev = sev
```
Comment (jeremyschlatter) [inline console.go]: This write races with the read in `Writer`. We should either make the read and write atomic or add a comment that this is not safe to call concurrently with `Writer`.

Comment (pquerna) [inline console.go]: I don't think a note, nor a atomic write is needed in this case.

strictly speaking, totally, yes. but in reality, a write to an int32 doesn't matter, and making all CPUs see the write at the same time (eg, actually atomic), doesn't actually matter.  we are flipping it from values of 0, 1, 2, 3.

**Comment by pquerna** [comment]
@jeremyschlatter go test fixed in 5e3637a

File: console.go (hunk 1 above)
Comment (jeremyschlatter) [inline console.go]: Fair enough.

**Comment by jeremyschlatter** [comment]
LGTM.

@r0mant any comments?

**Comment by r0mant** [comment]
lgtm

File: log.go [hunk 2]
```diff
@@ -5,7 +5,11 @@ import (
 	"io"
 )
 
-var loggers []Logger
+type grouplogger struct {
+	loggers []Logger
+}
+
+var gl grouplogger
```
Comment (horkhe) [inline log.go]: @pquerna just curious what was the point of introducing this?

**Pull request merged**